### PR TITLE
feat(route): add gogoanimehd

### PR DESCRIPTION
**Pull request by user4302**
## Involved Issue / 该 PR 相关 Issue

Close #

## Example for the Proposed Route(s) / 路由地址示例

```routes
/gogoanimehd/recent-releases
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [x] New Route / 新的路由
  - [x] Follows [v2 Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [v2 路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [x] Documentation / 文档说明
  - [x] EN / 英文文档
  - [ ] CN / 中文文档
- [ ] Full text / 全文获取
  - [ ] Use cache / 使用缓存
- [ ] Anti-bot or rate limit / 反爬/频率限制 
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施? 
- [ ] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [ ] Parsed / 可以解析
  - [ ] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

## Note / 说明
Get updates for recently released episodes that are added to gogoanimehd

### CODE REVIEW

**Review by TonyRL — COMMENTED** [review]
Missing documentation.
Guide: https://docs.rsshub.app/joinus/new-rss/add-docs

File: lib/v2/gogoanimehd/recent-releases.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        const formattedTitle = `<div>${title}<br/><h2>${episode}</h2></div>`;
+        const formattedDescription = `<img src='${img}' alt='${title}'>`;
+
+        const structuredData = {
+            title: formattedTitle,
+            description: formattedDescription,
+            link,
+        };
+        arrayOfItems.push(structuredData);
+    }
```
[the remark this answers is not in the document]
Comment (user4302) [reply lib/v2/gogoanimehd/recent-releases.js:32]: it dint work when i trued map,
ill try again.

[the remark this answers is not in the document]
Comment (JimenezLi) [reply lib/v2/gogoanimehd/recent-releases.js:32]: Did you use toArray() to convert Cheerio Object to Array?

[the remark this answers is not in the document]
Comment (user4302) [reply lib/v2/gogoanimehd/recent-releases.js:32]: oooh. ya i didnt use that, i used it now, its working perfectly.
thanks :D

**Comment by user4302** [comment]
> Missing documentation. Guide: https://docs.rsshub.app/joinus/new-rss/add-docs

hey, for this url
it asks to go to this location
`/website/i18n/en/docusaurus-plugin-content-docs/current/routes/programming.md.`
but the `current` directory does not exist, should i create `/current/routes` ?

https://github.com/DIYgod/RSSHub/tree/master/website/docs:
image

**Comment by JimenezLi** [comment]
You should add docs to `/website/docs/routes/anime.md`. Currently route docs (all languages) are mixed in `/website/docs/routes`.
Besides, your router is `/recent-releases`, but you are using `/recent-released` in your PR so route test says wrong path. You should edit the PR.

File: lib/v2/gogoanimehd/recent-releases.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    const recentReleases = $('.last_episodes');
+    const listItems = $(recentReleases).find('li');
+
+    const arrayOfItems = listItems.toArray().map((item) => {
+        const title = $(item).find('.name a').attr('title');
+        const episode = $(item).find('.episode').text();
+        const link = $(item).find('.name a').attr('href');
+        const img = $(item).find('.img a img').attr('src');
+
+        const formattedTitle = `<div>${title}<br/><h2>${episode}</h2></div>`;
```
Comment (TonyRL) [inline lib/v2/gogoanimehd/recent-releases.js:22]: Title only accepts a plain string.

Comment (user4302) [reply lib/v2/gogoanimehd/recent-releases.js:22]: oh :(
can i add the html and the episode value in the description ?

**Comment by user4302** [comment]
> You should add docs to `/website/docs/routes/anime.md`. Currently route docs (all languages) are mixed in `/website/docs/routes`. Besides, your router is `/recent-releases`, but you are using `/recent-released` in your PR so route test says wrong path. You should edit the PR.

oh i didnt see that, thanks. ill fix that too

**Comment by user4302** [comment]
could someone please let me know whats wrong with the jest test?
it says it expected -2 and got +2.
other than that im not sure whats failing in my code

**Comment by JimenezLi** [comment]
> could someone please let me know whats wrong with the jest test? it says it expected -2 and got +2. other than that im not sure whats failing in my code

The code doesn't have problem. Jest on Node 18 just fails itself.

**Comment by user4302** [comment]
> > could someone please let me know whats wrong with the jest test? it says it expected -2 and got +2. other than that im not sure whats failing in my code
> 
> The code doesn't have problem. Jest on Node 18 just fails itself.

nice :3

please let me know if theres anything else wrong, ill fix it asap

File: lib/v2/gogoanimehd/radar.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+module.exports = {
+    'gogoanimehd.io': {
+        _name: 'Gogoanime',
+        developer: [
+            {
+                title: 'Watch anime online, English anime online HD',
```
Comment (TonyRL) [inline lib/v2/gogoanimehd/radar.js:6]:
```suggestion
                title: 'Recent Releases',
```

**Comment by user4302** [comment]
Hi, this URL isn't working.
I get "error message: wrong path"

Will it take time for it to be deployed or something?

**Pull request merged**